internal/controller/velerobackup_predicate.go

```diff
@@ -0,0 +1,69 @@
+// velerobackup_predicate.go
+
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+func VeleroPredicate(scheme *runtime.Scheme) predicate.Predicate {
+	return nil
+}
+
+type VeleroBackupPredicate struct {
+	// We are watching only Velero Backup objects within
+	// namespace where OADP is.
+	OadpVeleroNamespace string
+	Logger              logr.Logger
+}
+
+func getBackupPredicateLogger(ctx context.Context, name, namespace string) logr.Logger {
+	return log.FromContext(ctx).WithValues("VeleroBackupPredicate", types.NamespacedName{Name: name, Namespace: namespace})
+}
+
+func (veleroBackupPredicate VeleroBackupPredicate) Create(ctx context.Context, evt event.CreateEvent) bool {
+	nameSpace := evt.Object.GetNamespace()
+	name := evt.Object.GetName()
+	log := getBackupPredicateLogger(ctx, name, nameSpace)
+	log.V(1).Info("Received Create VeleroBackupPredicate")
+
+	if nameSpace != veleroBackupPredicate.OadpVeleroNamespace {
+		return false
+	}
```

Comment:
```suggestion
	nameSpace := evt.Object.GetNamespace()
	if nameSpace != veleroBackupPredicate.OadpVeleroNamespace {
		return false
	}

	name := evt.Object.GetName()
	log := getBackupPredicateLogger(ctx, name, nameSpace)
	log.V(1).Info("Received Create VeleroBackupPredicate")
```

for performance, let's not call getName, getLogger, log if the namespace is different.